cli/resource/process.go

```diff
@@ -57,59 +84,58 @@ func ProcessFile(cli *cli.SensuCli, input string, recurse bool, processor Proces
 		if err != nil {
 			return err
 		}
-		resources, err := Parse(f)
+		res, err := Parse(f)
 		if err != nil {
 			return fmt.Errorf("in %s: %s", input, err)
 		}
-		if err := Validate(resources, cli.Config.Namespace()); err != nil {
-			return err
-		}
-		return processor.Process(cli.Client, resources)
+		resources = append(resources, res...)
+		return nil
 	})
+	return resources, err
 }
 
 // ProcessURL processes a url.
-func ProcessURL(cli *cli.SensuCli, client *http.Client, urly *url.URL, input string, recurse bool, processor Processor) error {
+func ProcessURL(client *http.Client, urly *url.URL, input string, recurse bool) ([]*types.Wrapper, error) {
+	var resources []*types.Wrapper
 	req, err := http.NewRequest("GET", urly.String(), nil)
 	if err != nil {
-		return err
+		return resources, err
 	}
 	resp, err := client.Do(req)
 	if err != nil {
-		return err
+		return resources, err
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode >= 400 {
 		buf := new(bytes.Buffer)
 		_, _ = io.Copy(buf, resp.Body)
-		return errors.New(buf.String())
+		return resources, errors.New(buf.String())
 	}
 
 	if strings.HasPrefix(resp.Header.Get("Content-Type"), "text/html") {
 		// The server returned us a directory listing
 		if !recurse {
-			return errors.New("use -r to enable directory recursion")
+			return resources, errors.New("use -r to enable directory recursion")
 		}
 		dec := xml.NewDecoder(resp.Body)
 		var dir httpDirectory
 		if err := dec.Decode(&dir); err != nil {
-			return err
+			return resources, err
 		}
 		for _, file := range dir.Files {
-			if err := Process(cli, client, filepath.Join(input, file), recurse, processor); err != nil {
-				return err
+			res, err := process(client, filepath.Join(input, file), recurse)
+			if err != nil {
+				return resources, err
 			}
+			resources = append(resources, res...)
+		}
+	} else {
+		resources, err := Parse(resp.Body)
```

Comment:
This is an ineffectual assignment, since it will create a new variable that shadows `resources` in the outer scope. 😬

Reply:
Shall I revert the changes made per https://github.com/sensu/sensu-go/pull/4017#discussion_r493655336?

Reply:
Sorry Nikki, my comment was confusing; either revert or don't shadow `resources`: `resources, err = Parse()`?

Reply:
I see the oops now hehe